src/index.tsx

```diff
@@ -238,7 +251,6 @@ const OTPInput = ({
               onKeyDown: handleKeyDown,
               onPaste: handlePaste,
               autoComplete: 'off',
-              maxLength: 1,
```

Comment:
Is it necessary to remove maxLength for this to work? Hoping it does not break other cases where you paste an arbitrary value

Reply:
Yes, unfortunately with the `maxLength` of 1 it is truncating the rest of the input. Are there some scenarios you'd like me to test? I'd be happy to help in any way I can.